### PR TITLE
feat(core): adopt editor features for journal

### PR DESCRIPTION
**Pull request by pengx17**

### CODE REVIEW

**Comment by pengx17** [comment]
Current dependencies on/for this PR:
* `canary`
  * **PR #5561** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5561?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #5657** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5657?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #5606** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5606?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #5638** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5638?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
      * **PR #5614** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5614?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/toeverything/AFFiNE/5638?utm_source=stack-comment).

File: packages/frontend/core/src/components/blocksuite/block-suite-editor/blocksuite-editor.tsx [hunk 1]
```diff
@@ -68,53 +71,51 @@ function usePageRoot(page: Page) {
...
     'pageId should exist for page reference'
   );
-  const referencedPage = meta.getPageMeta(
-    reference.delta.attributes.reference.pageId
-  );
-  const title = referencedPage?.title ?? 'not found';
-  let icon = <PageIcon />;
+  const pageId = reference.delta.attributes.reference.pageId;
+  const referencedPage = pageMetaHelper.getPageMeta(pageId);
+  const title = referencedPage?.title ?? 'Linked page not found';
```
Comment (CatsJuice) [inline packages/frontend/core/src/components/blocksuite/block-suite-editor/blocksuite-editor.tsx]: `TODO` for `i18n`?

File: packages/frontend/core/src/hooks/use-journal.ts [hunk 2]
```diff
@@ -82,25 +80,15 @@ export const useJournalHelper = (workspace: BlockSuiteWorkspace) => {
...
-   */
-  const openJournal = useCallback(
-    (maybeDate: MaybeDate) => {
-      const page = getJournalByDate(maybeDate);
-      navigateHelper.openPage(workspace.id, page.id);
+  const isPageTodayJournal = useCallback(
+    (pageId: string) => {
+      const date = dayjs().format(JOURNAL_DATE_FORMAT);
+      const d = adapter.getJournalPageDateString(pageId);
+      return isPageJournal(pageId) && d === date;
```
Comment (CatsJuice) [inline packages/frontend/core/src/hooks/use-journal.ts:87]: Maybe compare through `dayjs.isSame()`? 🤣 I still worry about different date formats.

Comment (pengx17) [reply packages/frontend/core/src/hooks/use-journal.ts:87]: It should be a disaster if we change the format in the future. I think we shall always stick to the 'yyyy-mm-dd' format and forget about it...

**Review by joooye34 — APPROVED** [review]

**Pull request merged**